quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCodeActions.java

```diff
@@ -111,6 +118,50 @@ private void doCodeActionsForUnknown(Diagnostic diagnostic, PropertiesModel docu
 		}
 	}
 
+	/**
+	 * Create code action fo 
+	 * @param diagnostic
+	 * @param document
+	 * @param projectInfo
+	 * @param valuesRulesManager
+	 * @param codeActions
+	 */
+	private void doCodeActionsForUnknownEnumValue(Diagnostic diagnostic, PropertiesModel document,
+			QuarkusProjectInfo projectInfo, ValuesRulesManager valuesRulesManager, List<CodeAction> codeActions) {
+		try {
+			PropertyValue propertyValue = (PropertyValue) document.findNodeAt(diagnostic.getRange().getStart());
+			PropertyKey propertyKey = ((Property) propertyValue.getParent()).getKey();
+			String value = propertyValue.getValue();
+			String propertyName = propertyKey.getPropertyName();
+
+			ExtendedConfigDescriptionBuildItem metaProperty = projectInfo.getPropertyFromName(propertyName);
+			if (metaProperty == null) {
+				return;
+			}
+
+			List<EnumItem> enums = metaProperty.getEnums();
+			if (enums == null || enums.isEmpty()) {
+				enums = valuesRulesManager.getValues(metaProperty, document);
+				if (enums == null || enums.isEmpty()) {
+					return;
+				}
+			}
+
+			for (EnumItem e : enums) {
+				if (isSimilar(e.getName(), value)) {

```

Comment:
In LSP4XML we have a current PR which does the same thing (but with XML Schema). The idea is to provide a code actions for similar names (like you have done). And if you have none similar name, you provide a code action per enum item by saying 'Replace with '

See https://github.com/angelozerr/lsp4xml/blob/655e17c197011cf41fe1880c54a861f2c1ef00c2/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/participants/codeactions/cvc_complex_type_2_4_aCodeAction.java#L76